lib/secure_headers/headers/content_security_policy.rb

```diff
@@ -214,5 +229,9 @@ def strip_source_schemes(source_list)
     def symbol_to_hyphen_case(sym)
       sym.to_s.tr("_", "-")
     end
+
+    def source_scheme(source)
+      source.match(/^([A-Za-z0-9\-\+.]+):\/\//)&.values_at(1)
```

Comment:
Is doing this using a regex the most reliable way? I imagine we had a good reason for not using a URI parser.